src/hooks/useCourtContracts.js

```diff
@@ -386,27 +419,34 @@ export function useTotalActiveBalancePolling(termId) {
 
   const timeoutId = useRef(null)
 
-  const fetchTotalActiveBalance = useCallback(() => {
-    timeoutId.current = setTimeout(() => {
-      return jurorRegistryContract
-        .totalActiveBalanceAt(termId)
-        .then(balance => {
-          setTotalActiveBalance(balance)
-          clearTimeout(timeoutId.current)
-          fetchTotalActiveBalance()
-        })
-        .catch(err => {
-          console.log(`Error fetching balance: ${err} retrying...`)
-          fetchTotalActiveBalance()
-        })
-    }, 1000)
-  }, [jurorRegistryContract, termId])
-
   useEffect(() => {
+    let cancelled = false
+
+    const fetchTotalActiveBalance = () => {
+      timeoutId.current = setTimeout(() => {
```

Comment:
Nitpick: there's not much point in using a ref now, since we've enclosed this within the `useEffect()`. We can just assign a local variable and clear it in the return.